File: java_implementation/src/main/java/com/tambapps/web/page_scrapping/saver/Saver.java
package com.tambapps.web.page_scrapping.saver;

import org.jsoup.nodes.Element;

import java.util.Optional;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Future;

public interface Saver {

  void printResult();

  Optional<Future> process(ExecutorService executor, Element element);

}

File: kotlin_implementation/src/main/kotlin/com/tambapps/kotlin/parameters/Saver.kt
package com.tambapps.kotlin.parameters

import org.jsoup.nodes.Element
import java.util.*
import java.util.concurrent.ExecutorService
import java.util.concurrent.Future

interface Saver {

    fun printResult()

    fun process(executor: ExecutorService, element: Element): Optional<Future<Any>>
}